In [5]:
import pandas as pd
import matplotlib.pyplot as plt
import psycopg2
from dotenv import load_dotenv
import os

print("Working Directory:", os.getcwd())

print("Loading environment variables from .env file...")
load_dotenv()

Working Directory: /home/guyil/aidoc-ehr-pipeline/notebooks
Loading environment variables from .env file...


True

### Connecting to the database

In [12]:
print("Connecting to the database...")
conn = psycopg2.connect(
    host="aidoc-ehr-pipeline.c0rumk4uq6lj.us-east-1.rds.amazonaws.com",
    dbname="aidoc_ehr",
    user="ehradmin",
    password=os.getenv("DB_PASSWORD")
)

if conn is not None:
    print("Connected to the database.")
else:
    raise AssertionError("Failed to connect to the database.")


Connecting to the database...
Connected to the database.


In [13]:
pd.read_sql("SELECT COUNT(*) FROM patient;", conn)

/tmp/ipykernel_265242/1794703786.py:1: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  pd.read_sql("SELECT COUNT(*) FROM patient;", conn)


,count
0,851


# EHR Pipeline — Exploratory Data Analysis

Synthetic FHIR patient data (Synthea, 851 patients) → structured Postgres schema.
This notebook explores the dataset before testing specific hypotheses about the
cardiac cohort: comorbidities, risk stratification, and care utilization.

## 1. Data Overview

In [21]:
row_query = """
SELECT 'patient' AS table_name, COUNT(*) FROM patient
UNION ALL
SELECT 'encounter', COUNT(*) FROM encounter
UNION ALL
SELECT 'condition', COUNT(*) FROM condition
UNION ALL
SELECT 'observation', COUNT(*) FROM observation
UNION ALL
SELECT 'procedure', COUNT(*) FROM procedure;
"""

row_counts = pd.read_sql(row_query, conn)

/tmp/ipykernel_265242/2682374938.py:13: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  row_counts = pd.read_sql(row_query, conn)


In [22]:
row_counts

,table_name,count
0,patient,851
1,condition,30321
2,encounter,49252
3,procedure,137415
4,observation,593045
